anydex/wallet/monero/xmr_wallet.py

```diff
@@ -0,0 +1,303 @@
+        payments = incoming_payments + outgoing_payments
+        return payments
+
+    def _normalize_transaction(self, payment: Payment) -> dict:
+        """
+        Take monero.transaction.Payment instance and turn it into Anydex dictionary format.
+
+        :return: formatted transaction dictionary
+        """
+        transaction = payment.transaction
+
+        if isinstance(payment, OutgoingPayment):
+            outgoing = True
+            from_ = payment.local_address
+            to = ''  # Monero hides recipient address
+        else:
+            outgoing = False
+            from_ = ''  # Monero hides sender address
+            to = payment.local_address
+
+        return {
+            'id': transaction.hash,
+            'outgoing': outgoing,
+            'from': from_,
+            'to': to,
+            'amount': payment.amount,
+            'fee_amount': transaction.fee,
+            'currency': self.get_identifier(),
+            'timestamp': time.mktime(transaction.timestamp.timetuple()),
+            'description': f'Confirmations: {transaction.confirmations}'
+        }
+
+    def get_incoming_payments(self, confirmed=True):
+        """
+        Get all confirmed incoming payments for this wallet.
+
+        :param: if set to True return only confirmed payments, else only mempool payments
+        :return: list of payments
+        """
+        if self._wallet_connection_alive():
+            return succeed(self.wallet.incoming(confirmed=confirmed, unconfirmed=(not confirmed)))
+        return fail(WalletConnectionError())
+
+    def get_outgoing_payments(self, confirmed=True):
+        """
+        Get all confirmed outgoing payments for this wallet.
+
+        :param: if True return only confirmed outgoing payments, else mempool payments
+        :return: list of payments
+        """
+        if self._wallet_connection_alive():
+            return succeed(self.wallet.outgoing(confirmed=confirmed, unconfirmed=(not confirmed)))
+        return fail(WalletConnectionError())
+
+    def min_unit(self):
+        # atomic unit: single piconero
+        return 1
+
+    def precision(self):
+        return 12
+
+    def get_identifier(self):
+        return 'XMR'
+
+    def get_confirmations(self, transaction: Payment):
+        """
+        Return number of confirmations transactions has received.
+
+        :param transaction: Payment object from monero-python library
+        :return: integer count
+        """
+        if self._wallet_connection_alive():
+            return succeed(self.wallet.confirmations(transaction))
+        return fail(WalletConnectionError())
+
+    def monitor_transaction(self, txid):
+        """
+        Blockchain is used to retrieve all transactions related to wallet.
+        No need for database and `monitor_transaction` method to store historical transactions.
+
+        :param txid: transaction id
+        """
+        super().monitor_transaction(txid)
```

Comment:
The monitor_transaction-method is not implemented for Monero because monitoring transactions in a database is not necessary for Monero. That's why I am calling the abstract method. Other choice could be `pass` or raising an exception. But I deemed this the best choice.

Maybe we should discuss this during our meeting.